### PR TITLE
YUI node handling replacement

### PR DESCRIPTION
**Pull request by huwshimi**
This is a big update (necessarily, once some code was updated it would cascade breakage) to replace the YUI methods for node selecting/creating etc.

It removes all instances (where possible) of:
- Y.one
- Y.all
- Y.Node.create
- getStyle
- setSyle
- addClass
- removeClass
- simulate
- Y.bind

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- pretty much just test the canvas as thoroughly as possible suggestions include:
- navigate to the canvas
- add a charm to the canvas
- the canvas should appear correctly
---
- add a charm to the canvas
- move the app around and make sure it can be dragged to all edges of the screen
---
- add a charm to the canvas
- click off the app
- hover over the app, the hover state should appear
---
- get the cookie notice to display
- check that you can close it
---
- add apache and wordpres
- create a relation
- the ambiguous relation menu should appear
- click on a relation type and it should get created
---
- drop a charm zip on the canvas
- drop an env.yaml on the canvas

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7095/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7096/

**Comment by hatched** [comment]
Lets not land this for Garak as that's being cut tomorrow.

**Comment by huwshimi** [comment]
@hatched yeah, good call.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7097/

File: jujugui/static/gui/src/app/views/topology/service.js [hunk 1]
```diff
@@ -498,14 +498,28 @@ YUI.add('juju-topology-service', function(Y) {
...
+    /**
       Attaches the drag and drop events for this view. These events need to be
       here because attaching them in the events object causes drag and drop
       events to stop bubbling at odd places cross browser.
 
       @method _attachDragEvents
     */
     _attachDragEvents: function() {
-      var container = this.get('container'),
+      var container = Y.one(this.get('container')),
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/topology/service.js]: Is the value of `this.get('container')` a dom node reference? If so I _think_ you want `Y.Node()` here not `Y.one()`

Comment (huwshimi) [inline jujugui/static/gui/src/app/views/topology/service.js]: That's correct, it appears to be able to attach the delegated events with either `Y.one` and `Y.Node` but I've updated it to `Y.Node` anyway.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7116/

**Comment by hatched** [comment]
QA OK

**Comment by hatched** [comment]
@huwshimi can you resolve these conflicts please then this one can land.

**Comment by hatched** [comment]
can land after the hotfix is cut :)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7152/

**Comment by huwshimi** [comment]
Thanks @Lukewh and @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**